core/src/main/resources/jenkins/model/Jenkins/404.properties

```diff
@@ -0,0 +1,2 @@
+notFound=The requested page wasn’t found. <br/>\
+  This means the page really doesn’t exist or a lack of permissions to discover the corresponding page.
```

Comment:
I don't think this message is helpful for the end user. If you lack permission to access a page, Jenkins doesn't yield a 404 to the end user, but displays the permission they're lacking.

Reply:
There are 2 permissions that are relevant when you access jobs/folders. The Item.READ permissions and the Item.DISCOVER permission. If you have READ you can access the job. If you have only DISCOVER you either get an accessdenied with the hint that you should log on (even when you are logged on). If you're not logged in, then you're redirected to the login page.
If you have neither permission you get a 404. 
So the message is accurate in my eyes